### PR TITLE
Upstream and improve webkit wasm tests

### PR DESCRIPTION
**Pull request by Ms2ger**
From https://trac.webkit.org/changeset/231194/webkit and https://trac.webkit.org/changeset/272221/webkit.

### CODE REVIEW

**Comment by nt1m** [comment]
@Ms2ger What is the status of this PR?

**Comment by Ms2ger** [comment]
I guess I need a review for the second commit, and then I'll land it.

File: wasm/webapi/wasm_stream_compile_test.html [hunk 1]
```diff
@@ -1,125 +1,115 @@
...
-      }) 
-      .then(stream => WebAssembly.compileStreaming(new Response(stream, { headers: { "Content-Type" : "application/wasm" }})));
-    assert_true(module instanceof WebAssembly.Module);
-  }, "compileStreaming receive response that deliver data by chunks as bufferArray");
+      const v = await fetch('resources/incrementer.wasm');
+      const buffer = await v.arrayBuffer();
+      const stream = new ReadableStream({
+        start(controller) {
+          (async () => {
+            await Promise.resolve().then(() => controller.enqueue(new Uint8Array(buffer.slice(0, 20))));
```
Comment (wingo) [inline wasm/webapi/wasm_stream_compile_test.html:74]: could this be just `await controller.enqueue(new Uint8Array(buffer.slice(0,20)))` (& likewise below)?

File: wasm/webapi/wasm_stream_instantiate_test.html [hunk 2]
```diff
@@ -1,121 +1,113 @@
...
-    assert_true(instance instanceof WebAssembly.Instance);
-  }, "instantiateStreaming receive response that deliver data by chunks as bufferArray");
+      const v = await fetch('resources/incrementer.wasm');
+      const buffer = await v.arrayBuffer();
+      const stream = new ReadableStream({
+        start(controller) {
+          (async () => {
+            await Promise.resolve().then(() => controller.enqueue(new Uint8Array(buffer.slice(0, 20))));
+            await Promise.resolve().then(() => controller.enqueue(new Uint8Array(buffer.slice(20, buffer.byteLength))));
+            await Promise.resolve().then(() => controller.close());
```
Comment (wingo) [inline wasm/webapi/wasm_stream_instantiate_test.html:75]: similar question here

**Review by wingo — APPROVED** [review]

File: wasm/webapi/wasm_stream_compile_test.html (hunk 1 above)
Comment (Ms2ger) [reply wasm/webapi/wasm_stream_compile_test.html:74]: That seems to work as well, but I don't know if it changes timing / the goal of the test

**Pull request merged**